crates/storage/provider/src/providers/database/provider.rs

```diff
@@ -141,9 +142,8 @@ impl<TX: DbTxMut> DatabaseProvider<TX> {
     }
 }
 
-impl<TX: DbTx + 'static> DatabaseProvider<TX> {
-    /// Storage provider for state at that given block
-    pub fn state_provider_by_block_number(
+impl<TX: DbTx + 'static> TryIntoHistoricalStateProvider for DatabaseProvider<TX> {
+    fn try_into_history_at_block(
```

Comment:
ah we need to consume this type here, hence this accepts self.

I believe this is special case because the blockchain-tree operates on the providerfactory directly.